frontends/p4/unusedDeclarations.h

```diff
@@ -144,15 +183,15 @@ class RemoveUnusedDeclarations : public Transform {
  * IR::P4Table or IR::Declaration_Instance is removed.
  */
 class RemoveAllUnusedDeclarations : public PassRepeated {
+    UsedDeclSet used;
+
  public:
-    RemoveAllUnusedDeclarations(ReferenceMap *refMap, const RemoveUnusedPolicy &policy,
-                                bool warn = false)
-        : PassManager({new ResolveReferences(refMap),
-                       policy.getRemoveUnusedDeclarationsPass(refMap, warn)}) {
+    explicit RemoveAllUnusedDeclarations(const RemoveUnusedPolicy &policy, bool warn = false)
+        : PassManager({new CollectUsedDeclarations(used),
+                       policy.getRemoveUnusedDeclarationsPass(used, warn)}) {
         // Unused extern instances are not removed but may still trigger
         // warnings.  The @warned set keeps track of warnings emitted in
         // previous iterations to avoid emitting duplicate warnings.
```

Comment:
I think this comment belongs either to `RemoveUnusedDeclarations` class, or maybe to the `warned` member, but I don't see what it is for here, when `warned` does not appear here.

Reply:
Yeah, likely it left after some earlier refactoring. I'll fix it while here.